### ISSUE TITLE
Singularity 3.0.1-1.el6 dependent on the host session's environment

### ISSUE THREAD

**Issue by mboisson** [apptainer/singularity#2445]
## Version of Singularity:
3.0.1-1.el6

### Expected behavior

Starting singularity should be pretty much independent on the session's environment

### Actual behavior

Running singularity depends on the session's environment. 

### Steps to reproduce behavior

```
$ /opt/software/singularity-3.0/bin/singularity --version
singularity version 3.0.1-1.el6
``` 

```
$ /opt/software/singularity-3.0/bin/singularity shell --nv pytorch-18.01-py3.simg
WARNING: ldconfig execution error: exit status 1
INFO:    Unable to capture nvidia bind points: exit status 1
ERROR  : Failed to read JSON configuration from socket: Bad file descriptor
``` 

On the same system : 

```
$ /usr/bin/env -i PATH=/usr/bin:/usr/local/sbin:/usr/sbin:/sbin /opt/software/singularity-3.0/bin/singularity shell --nv pytorch-18.01-py3.simg
WARNING: underlay of /usr/bin/nvidia-smi required more than 50 (323) bind mounts
Singularity pytorch-18.01-py3.simg:~>
``` 

This is caused by modules loaded and various environment variables that are set on a cluster.

**Comment by mboisson** [comment]
This looks similar to #2060, but obviously, this is not fixed in 3.0.1 since that is what we have installed.

**Comment by mboisson** [comment]
For what it's worth, we have confirmed that the version that we compiled has 
```
func ExecuteSingularity() {
  defaultEnv := "/bin:/usr/bin:/sbin:/usr/sbin:/usr/local/bin:/usr/local/sbin"

  // backup user PATH
  userEnv := strings.Join([]string{os.Getenv("PATH"), defaultEnv}, ":")
  os.Setenv("USER_PATH", userEnv)

  os.Setenv("PATH", defaultEnv)
  if err := SingularityCmd.Execute(); err != nil {
    os.Exit(1)
  }
}
```

**Comment by mboisson** [comment]
@jscook2345, actually, the error message the OP is two separate problems. The `ldconfig` error message is related to the `PATH`. I can workaround it by sanitizing the `PATH` before calling `singularity`. The JSON error is related to the overall size of the environment.

**Comment by mboisson** [comment]
This seems to have been resolved in 3.1.
